addons/mail/models/mail_thread.py

```diff
@@ -219,12 +219,19 @@ def _search_message_needaction(self, operator, operand):
     def _compute_message_has_error(self):
         res = {}
         if self.ids:
-            self._cr.execute(""" SELECT msg.res_id, COUNT(msg.res_id) FROM mail_message msg
-                                 RIGHT JOIN mail_notification rel
-                                 ON rel.mail_message_id = msg.id AND rel.notification_status in ('exception','bounce')
-                                 WHERE msg.author_id = %s AND msg.model = %s AND msg.res_id in %s AND msg.message_type != 'user_notification'
-                                 GROUP BY msg.res_id""",
-                             (self.env.user.partner_id.id, self._name, tuple(self.ids),))
+            self._cr.execute("""
+                    SELECT msg.res_id, COUNT(msg.res_id)
+                      FROM mail_message msg
+                INNER JOIN mail_notification notif
+                        ON notif.mail_message_id = msg.id
+                     WHERE notif.notification_status in ('exception','bounce')
+                       AND notif.author_id = %s
+                       AND msg.author_id = %s
+                       AND msg.model = %s
+                       AND msg.res_id in %s
+                       AND msg.message_type != 'user_notification'
+                  GROUP BY msg.res_id
+            """, (self.env.user.partner_id.id, self.env.user.partner_id.id, self._name, tuple(self.ids),))
```

Comment:
Can't you use named parameters ? When having lot of parameters it makes everything easier to read ...

Reply:
Cool :) as anyway you are somehow "breaking history" (hard to see pre / post history), let us do it cleanly directly ^^ .